### PR TITLE
Add rate limit to bridging

### PR DESCRIPTION
**Pull request by danhper**
Add rate limit to bridging. This uses a model similar to Chainlink where each "lane" has a maximum capacity that refills at a certain speed.
We'll need to decide on the capacity and the refill speed.

### CODE REVIEW

**Review by aklamun — APPROVED** [review]

File: src/CCIPHelpers.sol [hunk 1]
```diff
@@ -43,4 +45,25 @@ library CCIPHelpers {
...
   }
+
+  function validateRateLimit(
+    uint64 chainSelector,
+    uint256 amount,
+    IGydBridge.RateLimitData memory rateLimit,
+    IGydBridge.ChainMetadata memory chainMeta
+  ) internal view returns (IGydBridge.RateLimitData memory) {
+    uint256 ellapsedSinceReplenish = block.timestamp - rateLimit.lastRefill;
+    uint256 amountToReplenish = ellapsedSinceReplenish * chainMeta.refillRate;
```
[the remark this answers is not in the document]
Comment (danhper) [reply src/CCIPHelpers.sol:56]: That's one less than I thought

File: script/DeployGydL1CCIPEscrow.s.sol [hunk 2]
```diff
@@ -38,6 +38,10 @@ contract DeployGydL1CCIPEscrow is Script {
 
   uint256 gasLimit = 200_000; // max 200k gas to complete the bridging
 
+  uint256 capacity = 100_000; // max 100k GYD at once
```
[the remark this answers is not in the document]
Comment (danhper) [reply script/DeployGydL1CCIPEscrow.s.sol:41]: This is a dummy value but we definitely need to scale this here!

File: script/DeployGydL1CCIPEscrow.s.sol [hunk 3]
```diff
@@ -38,6 +38,10 @@ contract DeployGydL1CCIPEscrow is Script {
...
+
+  uint256 refillRate = 10; // 1 GYD per second
```
[the remark this answers is not in the document]
Comment (danhper) [reply script/DeployGydL1CCIPEscrow.s.sol:43]: I'll share something with the values that we'll actually use!

[the remark this answers is not in the document]
Comment (sschuldenzucker) [reply script/DeployGydL1CCIPEscrow.s.sol:43]: Also, this would be 1 GYD per second, not 10. :)

**Review by sschuldenzucker — APPROVED** [review]
LGTM

**Pull request merged**